web/views.py

```
@@ -6929,3 +6933,207 @@ def delete_post(request, post_id):
+        goal.priority = request.POST.get('priority')
+        goal.save()
+        
+        messages.success(request, 'Study goal updated successfully!')
+        return redirect('study_plan_detail', plan_id=goal.study_plan.id)
+    
+    context = {'goal': goal}
+    return render(request, 'study_planner/update_goal.html', context)
+
+
+@login_required
+def delete_study_plan(request, plan_id):
```

Ruff (0.8.2)

Missing return type annotation for public function `delete_study_plan`

(ANN201)